src/daft-connect/src/convert/plan_conversion.rs

```diff
@@ -0,0 +1,134 @@
+use std::{collections::HashSet, sync::Arc};
+
+use daft_logical_plan::{LogicalPlanBuilder, ParquetScanBuilder};
+use eyre::{bail, eyre, Result, WrapErr};
+use spark_connect::{
+    expression::Alias,
+    read::{DataSource, ReadType},
+    relation::RelType,
+    Filter, Read, Relation, WithColumns,
+};
+use tracing::warn;
+
+use crate::convert::expression;
+
+pub fn to_logical_plan(plan: Relation) -> Result<LogicalPlanBuilder> {
+    let scope = std::thread::spawn(|| {
+        let rel_type = plan.rel_type.ok_or_else(|| eyre!("rel_type is None"))?;
+
+        match rel_type {
+            RelType::ShowString(..) => {
+                bail!("ShowString is only supported as a top-level relation")
+            }
+            RelType::Filter(filter) => parse_filter(*filter).wrap_err("parsing Filter"),
+            RelType::WithColumns(with_columns) => {
+                parse_with_columns(*with_columns).wrap_err("parsing WithColumns")
+            }
+            RelType::Read(read) => parse_read(read),
+            _ => bail!("Unsupported relation type: {rel_type:?}"),
+        }
+    });
+
+    scope.join().unwrap()
+}
+
+fn parse_filter(filter: Filter) -> Result<LogicalPlanBuilder> {
```

Comment:
nit: i don't think were parsing the filter, but instead attempting to plan/convert it to our logical filter.